From 7464ca491d7e22b0de0cf0dfb925037ec1d4edab Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval <msandova@gnome.org>
Date: Thu, 21 Mar 2024 22:17:00 +0100
Subject: [PATCH 3/3] sorting: Do not declare None == not None

Right now we declare two things as equal if one is missing a date.
---
 gsecrets/sorting.py | 22 +++++++++++++++-------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/gsecrets/sorting.py b/gsecrets/sorting.py
index 688c9bf0..c043cc5c 100644
--- a/gsecrets/sorting.py
+++ b/gsecrets/sorting.py
@@ -45,19 +45,27 @@ class SortingHat:
     def sort_by_ctime_asc(
         ele1: SafeElement, ele2: SafeElement, _data: object = None
     ) -> int:
-        if ele1.ctime is None or ele2.ctime is None:
-            return 0
+        if ele1.ctime is not None and ele2.ctime is not None:
+            return ele1.ctime.compare(ele2.ctime)
+        if ele1.ctime is None and ele2.ctime is not None:
+            return 1
+        if ele1.ctime is not None and ele2.ctime is None:
+            return -1
 
-        return ele1.ctime.compare(ele2.ctime)
+        return 0
 
     @staticmethod
     def sort_by_ctime_dec(
         ele1: SafeElement, ele2: SafeElement, _data: object = None
     ) -> int:
-        if ele1.ctime is None or ele2.ctime is None:
-            return 0
-
-        return ele2.ctime.compare(ele1.ctime)
+        if ele1.ctime is not None and ele2.ctime is not None:
+            return ele2.ctime.compare(ele1.ctime)
+        if ele1.ctime is None and ele2.ctime is not None:
+            return -1
+        if ele1.ctime is not None and ele2.ctime is None:
+            return 1
+
+        return 0
 
 
 SortingHat.sort_funcs = {
-- 
2.44.0

